integration_tests/test_domestic_migration.py

```diff
@@ -16,12 +16,39 @@ def test_individuals_move(simulants_on_adjacent_timesteps):
 
 def test_individuals_move_into_new_households(simulants_on_adjacent_timesteps):
     for before, after in simulants_on_adjacent_timesteps:
-        new_household_movers = (
+        # NOTE: This set is not exactly the same as "new-household movers," as
+        # implemented in the component, because it can in rare cases include
+        # non-reference-person movers who join a household that was *just*
+        # established by a new-household mover (within the same time step).
+        movers_into_new_households = (
             (before["household_id"] != after["household_id"])
             & (after["housing_type"] == "Standard")
             & (~after["household_id"].isin(before["household_id"]))
         )
+        assert movers_into_new_households.any()
+
+        assert (
+            after[movers_into_new_households]["relation_to_household_head"]
+            # Handling the non-reference-person movers described above.
+            .isin(["Reference person", "Other nonrelative"]).all()
+        )
+
+        # These are the true new-household movers, as that term is used in the
+        # migration component: the movers who establish a new household.
+        new_household_movers = movers_into_new_households & (
+            after["relation_to_household_head"] == "Reference person"
+        )
         assert new_household_movers.any()
+        # There is exactly one new-household mover for each new household.
+        assert (
+            new_household_movers[movers_into_new_households]
+            .groupby(after[movers_into_new_households]["household_id"])
+            .sum()
+            == 1
+        ).all()
+        # These should be the vast majority, since non-reference-person moves to
+        # new households will be quite rare.
+        assert new_household_movers.sum() / movers_into_new_households.sum() > 0.95
```

Comment:
I like the sentiment here, but is 95% a reasonable number?

Reply:
I'd think this number should scale with the size of the sim population. If we have fewer than 20 new households formed, a single person moving into one of these new households will cause this test to fail. We really don't want false positives in our tests, since CRNs means we'll fail every time if we fail at all. 

That being said, I don't think it's important to think too hard about what the number should be in this case. If this passes on the sizes we're using, that's fine. We're really just checking that we don't have everyone moving into these new households for some reason.

Reply:
@rmudambi Good point, I didn't think about small population sizes. I think that in general we should consider 20k a minimum -- for very small sample sizes, I think there are other tests that also fail.